dashboard/lib/service/appengine_cocoon.dart

```diff
@@ -167,12 +167,10 @@ class AppEngineCocoonService implements CocoonService {
         }));
     final bool success = response.statusCode == HttpStatus.ok;
```

Comment:
Prefer not nesting code, as it makes it harder to read.

Additionally, assume if HTTP 200 was returned that there was no error

```dart
if (response.statusCode == HttpStatus.ok) {
  return const CocoonResponse<bool>.data(true);
}

return CocoonResponse<bool>.error('HTTP ${response.statusCode}: ${response.body}');
```

Reply:
For sure, I will rewrite the logic. Just a few contexts on this block, it is hoping to addresses the feedback that if a task has already succeeded and could not be rerun, instead of showing nothing or other notification, we now prompt a notification to say succeeded task cannot rerun. It checks that aside from 200 code, whether the returned response body is empty. If it's empty then it's a task that has already succeeded and could not be rerun.

Reply:
I don't think you can rely on the response. For example, in https://github.com/flutter/cocoon/pull/1610 is removing that logic, and it'll return an error when trying to rerun a task that shouldn't be rerun

Reply:
umm I might be wrong, but I thought 1610 kept L76-83 (https://github.com/flutter/cocoon/pull/1610/files#diff-8b18660d0517d62995b30d9277a6cd6601f428ccd5def5d4535f668cd6ba8cb3L76-L83), and my understanding is that this part will be responsible for returning an empty response body if task has already succeeded?

Reply:
No, it'll return an exception in that case (it's being moved in checkRerunBuilder)